### ISSUE TITLE
Have Events run off native handler

### ISSUE THREAD

**Issue by michael-watson** [michael-watson/particle-xamarin#7]
Modernhttpclient was locking up the UI thread when running the continuous stream from HttpClient. 

Need to implement native handlers with a proper Task system

**Comment by michael-watson** [comment]
There is a known issue about this in modernhtttpclient and it has been open since June of last year.
https://github.com/paulcbetts/ModernHttpClient/issues/163

Going to need to inspect the default httpclient handler and compare with modernhttpclient iOS implementation. Most likely will need to Fork the project and debug from there.

**Comment by justmobilize** [comment]
I'm also going to not that if your using System.Net.Http for Windows on events it won't work. It will close the connection after the first bit of data. You'll need to use a HttpListener.

**Comment by michael-watson** [comment]
Interesting @justmobilize, I was able to get Events working for a Windows 10 UWP application here using System.Net.Http. The application was used at our mobile conference Evolve and seemed to be successful. I will have to double check all of this in that sample and will post information when I get it. Have you tried it out in that sense?

On vacation next week, but I'll be back on after to check this out.

**Comment by justmobilize** [comment]
@michael-watson That is true. So it depends on who your targeting. For Windows Phone Devices, Windows 10 Phone [has less than 10% of the market](http://www.slideshare.net/adduplex/adduplex-windows-phone-statistics-report-april-2016) and it isn't going up very fast. And that method doesn't work in Windows 8.1.
